dio/lib/src/options.dart

```diff
@@ -352,10 +361,19 @@ class Options {
 
   /// HTTP request headers.
   ///
-  /// The keys of the header are case-insensitive,
-  /// e.g.: content-type and Content-Type will be treated as the same key.
+  /// The equality of the header keys is case-insensitive,
```

Comment:
What about the following:

By default header keys are compared without taking the case into consideration, e.g.: `content-type` and `Content-Type` will be treated as the same key. To change this set [caseSensitiveHeaders] to [true]

Reply:
The fact is here we don't use case-sensitive map even if the flag is true. Otherwise it will be too complicated to handle reading fields like `Headers[Headers.contentTypeHeader]` which takes massive conversion if we do this.

Reply:
Ohh, I missed that. Okay, then just disregard this comment